### PR TITLE
Notifications

### PR DESCRIPTION
**Pull request by huseinkareem**

### CODE REVIEW

File: WalkingChallenge/LocalizedStrings.swift [hunk 1]
```diff
@@ -92,6 +92,17 @@ struct Strings {
     static let joinChallenge: String =
         NSLocalizedString("Join Challenge", comment: "Join Challenge")
   }
+
+  struct Notifications {
+    static let title =
+      NSLocalizedString("Notifications", comment: "Notifications")
+    static let notificationCellTitleLabelPlaceholderTextFormat: String =
+        NSLocalizedString("%@ %@ has joined your team!",
```
Comment (compnerd) [inline WalkingChallenge/LocalizedStrings.swift:100]: Do we really want to display this as `<FirstName> <LastName>`?  Do certain regions not have alternative display formats?

File: WalkingChallenge/Notifications/NotificationTableViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,54 @@
+//
+//  NotificationTableViewCell.swift
+//  WalkingChallenge
+//
+//  Created by Husein Kareem on 4/12/19.
+//  Copyright © 2019 AKDN. All rights reserved.
+//
```
Comment (compnerd) [inline WalkingChallenge/Notifications/NotificationTableViewCell.swift:7]: Please use the same header as the rest of the code (it is the copyright blob)

File: WalkingChallenge/Notifications/NotificationTableViewCell.swift [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+import Foundation
+import UIKit
+import SnapKit
+
+class NotificationTableViewCell: UITableViewCell {
+
```
Comment (compnerd) [inline WalkingChallenge/Notifications/NotificationTableViewCell.swift:14]: Unnecessary whitespace

File: WalkingChallenge/Notifications/NotificationTableViewCell.swift [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  static let CellIdentifier: String = "NotificationCell"
+
+  lazy var notificationInfoLabel: UILabel = UILabel()
+  lazy var notificationDateLabel: UILabel = UILabel()
+
+  override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+    super.init(style: style, reuseIdentifier: reuseIdentifier)
+
+    self.backgroundColor = Style.Colors.DeepTurquoise.withAlphaComponent(0.0735)
```
Comment (compnerd) [inline WalkingChallenge/Notifications/NotificationTableViewCell.swift:23]: Where was this color selected?  We should be trying to stick to the color style guide

File: WalkingChallenge/Notifications/NotificationTableViewCell.swift [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    self.layer.masksToBounds = false
+    self.layer.renderShadow()
+
+    self.addSubview(self.notificationInfoLabel)
+    self.addSubview(self.notificationDateLabel)
+
+    self.notificationInfoLabel.snp.makeConstraints { (make) in
+      make.top.equalTo(self).offset(10)
```
Comment (compnerd) [inline WalkingChallenge/Notifications/NotificationTableViewCell.swift:31]: Please use the `Style` constants for the offset (although, if there isn't one for 10 … we should see if there are other places that use that and if so create it else leave it be)

File: WalkingChallenge/Notifications/NotificationTableViewCell.swift [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    self.notificationInfoLabel.numberOfLines = 0
+
+    self.notificationDateLabel.snp.makeConstraints { (make) in
+      make.top.equalTo(self.notificationInfoLabel.snp.bottom).offset(Style.Padding.p16)
+      make.bottom.equalToSuperview().inset(Style.Padding.p16)
+      make.left.right.equalToSuperview().inset(Style.Padding.p24)
+    }
+
+    //TODO: Placeholder, use name from data
+    self.notificationInfoLabel.text = String(format: Strings.Notifications.notificationCellTitleLabelPlaceholderTextFormat,
```
Comment (compnerd) [inline WalkingChallenge/Notifications/NotificationTableViewCell.swift:44]: We already have the way to get the user's name, please use that

File: WalkingChallenge/Notifications/NotificationTableViewCell.swift [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+                                             arguments: ["FirstName", "LastName"])
+    //TODO: Placeholder for now, use date (w/ formatter) from data
+    self.notificationDateLabel.text = "01/01/2000"
```
Comment (compnerd) [inline WalkingChallenge/Notifications/NotificationTableViewCell.swift:47]: We already have a date formatter utility.  What is this date supposed to be?

File: WalkingChallenge/Notifications/Notifications.swift [hunk 8]
```diff
@@ -29,6 +29,92 @@
 
 import UIKit
 import Foundation
+import SnapKit
 
 class Notifications: UIViewController {
+
```
Comment (compnerd) [inline WalkingChallenge/Notifications/Notifications.swift:35]: Unnecessary whitespace

File: WalkingChallenge/Notifications/Notifications.swift [hunk 9]
```diff
@@ -29,6 +29,92 @@
...
+                                                   for: indexPath) as? NotificationTableViewCell else {
+      return UITableViewCell()
+    }
+
+    return cell
+  }
+}
+
+extension Notifications: UITableViewDelegate {
+
```
Comment (compnerd) [inline WalkingChallenge/Notifications/Notifications.swift:116]: Unnecessary whitespace

**Review by compnerd — CHANGES_REQUESTED** [review]
Please squash the changes into a single commit.

File: WalkingChallenge/LocalizedStrings.swift (hunk 1 above)
Comment (huseinkareem) [reply WalkingChallenge/LocalizedStrings.swift:100]: I'm not sure. Maybe we can leave it until we get the response data?

File: WalkingChallenge/Notifications/NotificationTableViewCell.swift (hunk 4 above)
Comment (huseinkareem) [reply WalkingChallenge/Notifications/NotificationTableViewCell.swift:23]: It's from the Marvel design.

File: WalkingChallenge/Notifications/NotificationTableViewCell.swift (hunk 7 above)
Comment (huseinkareem) [reply WalkingChallenge/Notifications/NotificationTableViewCell.swift:47]: I'm using it as a placeholder until we set it from the notification response data.

File: WalkingChallenge/Notifications/NotificationTableViewCell.swift (hunk 5 above)
Comment (huseinkareem) [reply WalkingChallenge/Notifications/NotificationTableViewCell.swift:31]: nope, there isn't one for 10 and this is the only place i'm using it. maybe i can use 8 or 12 or i can leave it.

File: WalkingChallenge/Notifications/NotificationTableViewCell.swift (hunk 4 above)
Comment (compnerd) [reply WalkingChallenge/Notifications/NotificationTableViewCell.swift:23]: I don't see turquoise in the mockup.  Got a link?
